rr_iarrc/src/cone_detection/cone_detection_bottom.cpp

```diff
@@ -0,0 +1,133 @@
+#include <ros/ros.h>
+#include <ros/publisher.h>
+#include <cv_bridge/cv_bridge.h>
+#include <sensor_msgs/Image.h>
+#include <opencv2/imgproc/imgproc.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/opencv.hpp>
+#include <opencv2/imgcodecs.hpp>
+#include <opencv2/opencv.hpp>
+#include <iostream>
+#include <stdlib.h>
+#include <stdio.h>
+
+using namespace std;
+
+cv::Mat kernel(int, int);
+void cutEnvironment(const cv::Mat&);
+void publishMessage(ros::Publisher,const cv::Mat&, std::string);
+cv::Mat overlayBinaryGreen(cv::Mat&, const cv::Mat&);
+cv_bridge::CvImagePtr cv_ptr;
+
+ros::Publisher pub, pub1, pub2;
+int low_H, high_H, low_S, low_V;
+int blockSky_height, blockWheels_height, blockBumper_height;
+
+/**
+ * Gets the bottom edge of the cones
+ * @param msg image from camera
+ */
+void img_callback(const sensor_msgs::ImageConstPtr& msg) {
+    //Convert msg to Mat image
+    cv_ptr = cv_bridge::toCvCopy(msg, "bgr8");
+    cv::Mat frame = cv_ptr->image;
+
+    int originalHeight = frame.rows;
+    int originalWidth = frame.cols;
+    cv::resize(frame, frame, cv::Size(400, 400));
+
+    //Get Orange-HSV Cones
+    cv::Mat hsv_frame, orange_found, orange_found_rot, nonZeroCoordinates, green_lines;
+    cv::cvtColor(frame, hsv_frame, cv::COLOR_BGR2HSV);
+    cv::inRange(hsv_frame, cv::Scalar(low_H, low_S, low_V), cv::Scalar(high_H, 255, 255), orange_found);
+    cutEnvironment(orange_found);
+
+    //Rotates Image because findNonZero() gives points already sorted by Y-coordinate
```

Comment:
I am confused why you rotate this image or the `last` variable. I suspect you may be expected behavior from `cv::findNonZero()` that is not guaranteed (giving points from top to bottom). If that is the case, just make function to iterate from the bottom of each column until you hit a white pixel, then add that to your image and continue.